src/backtrace/dbghelp.rs

```diff
@@ -90,12 +90,36 @@ struct MyContext(CONTEXT);
 
 #[inline(always)]
 pub unsafe fn trace(cb: &mut dyn FnMut(&super::Frame) -> bool) {
+    let thread = GetCurrentThread();
+    trace_thread(cb, thread)
+}
+
+#[inline(always)]
+pub unsafe fn trace_thread(cb: &mut dyn FnMut(&super::Frame) -> bool, thread: HANDLE) {
     // Allocate necessary structures for doing the stack walk
     let process = GetCurrentProcess();
-    let thread = GetCurrentThread();
 
     let mut context = mem::zeroed::<MyContext>();
-    RtlCaptureContext(&mut context.0);
+    if thread == GetCurrentThread() || thread.is_null() {
+        RtlCaptureContext(&mut context.0);
+    } else {
+        // The suspending and resuming of threads is very risky.
+        // It can end in a deadlock if the current thread tries to access
+        // an object thats been locked by the suspended thread.
+        // That's why we only do as little work as possible while
+        // the thread is suspended, and resume it quickly after.
+        // There might be more pitfalls we haven't thought of or encountered
```

Comment:
No speculative "maybe this is bad?" in fresh code, please. I'd like to hear why this is actually okay, despite concerns about corrupting the backtrace.